src/rgw/rgw_sal.h

```diff
@@ -675,11 +693,11 @@ class Bucket {
 				 const bucket_index_layout_generation& idx_layout,
 				 int shard_id, RGWGetBucketStats_CB* ctx) = 0;
     /** Sync this bucket's stats to the owning user's stats in the backing store */
-    virtual int sync_user_stats(const DoutPrefixProvider *dpp, optional_yield y) = 0;
-    /** Refresh the metadata stats (size, count, and so on) from the backing store */
-    virtual int update_container_stats(const DoutPrefixProvider* dpp, optional_yield y) = 0;
+    virtual int sync_user_stats(const DoutPrefixProvider *dpp, optional_yield y,
+                                RGWBucketEnt* ent) = 0;
```

Comment:
I'd prefer to see the `ent` parameter documented that it's optional, and that if provided it will contain the stats for the bucket. Is there a reason why you did not set `nullptr` as the default value?

Reply:
in general, i think that non-optional output parameters should always be references instead of pointers, but our codebase isn't consistent about this. do you think the parameter name `optional_ent` would be enough to clarify this here, or would you prefer a comment?

regarding default arguments on virtual functions specifically, i think they're error-prone because the compiler doesn't guarantee that the base class and its overrides are consistent. so you might get a different default depending on what type of pointer you call the function on

according to https://google.github.io/styleguide/cppguide.html#Default_Arguments:
> Default arguments are allowed on non-virtual functions when the default is guaranteed to always have the same value.